### PR TITLE
Copy lit tests from `tritongpu_to_llvm.mlir` to `tritongpu_to_gen.mlir`

### LINKED ISSUE #756: Copy lit tests from `tritongpu_to_nvvm.mlir` to `tritongpu_to_gen.mlir`
Copy test `test/Conversion/tritongpu_to_nvvm.mlir` to `test/Conversion/tritongpu_to.gen.mlir` when applicable on XPU.

### PR DESCRIPTION
**Pull request by anmyachev**
Closes #756

### CODE REVIEW

File: test/Conversion/intel/tritongpu_to_gen.mlir [hunk 1]
```diff
@@ -1655,6 +1655,7 @@ module attributes {"triton_gpu.target" = "xpu:DEVICE_ARCH.PVC", "triton_gpu.num-
 #blocked0 = #triton_gpu.blocked<{sizePerThread = [2], threadsPerWarp = [32], warpsPerCTA = [4], order = [0], CTAsPerCGA = [1], CTASplitNum = [1], CTAOrder = [0]}>
 module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 4 : i32} {
   // CHECK-LABEL: abs_is_int_min_poison
+  // CHECK: %{{.*}} = "llvm.intr.abs"(%{{.*}}) <{is_int_min_poison = true}> : (i32) -> i32
```
Comment (anmyachev) [inline test/Conversion/intel/tritongpu_to_gen.mlir:1658]: Shouldn't it be like `is_int_min_poison = false` as in `tritongpu_to_llvm.mlir`?

File: test/Conversion/intel/tritongpu_to_gen.mlir [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// RUN: triton-opt %s -split-input-file --intel-allocate-shared-memory  --convert-triton-intel-gpu-to-llvm | FileCheck %s --implicit-check-not=llvm.inline_asm
+// RUN: triton-opt %s -split-input-file --intel-allocate-shared-memory  --convert-triton-intel-gpu-to-llvm | FileCheck %s --implicit-check-not=llvm.inline_asm --dump-input fail --dump-input-context 5
```
Comment (anmyachev) [inline test/Conversion/intel/tritongpu_to_gen.mlir:1]: will revert these changes after the review.

**Comment by anmyachev** [comment]
@whitneywhtsang could you take a look?

File: test/Conversion/intel/tritongpu_to_gen.mlir (hunk 1 above)
Comment (whitneywhtsang) [reply test/Conversion/intel/tritongpu_to_gen.mlir:1770]: Can you open an issue to investigate?

Comment (anmyachev) [reply test/Conversion/intel/tritongpu_to_gen.mlir:1770]: Fix in https://github.com/intel/intel-xpu-backend-for-triton/pull/1701

File: test/Conversion/intel/tritongpu_to_gen.mlir
[the remark this answers is not in the document]
Comment (anmyachev) [reply test/Conversion/intel/tritongpu_to_gen.mlir:1]: > Should we also port test_local_store and test_local_store_subview?

I can't do this because of an error: ` error: failed to legalize operation 'triton_gpu.local_store' that was explicitly marked illegal`. Should I file an issue for this problem?

@whitneywhtsang ready for review :)

File: test/Conversion/intel/tritongpu_to_gen.mlir [hunk 3]
```diff
@@ -951,6 +1008,7 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 4 :
   tt.func @matmul_fmadot(%ptr:!tt.ptr<f32> {tt.divisibility = 16 : i32},
```
[the remark this answers is not in the document]
Comment (anmyachev) [reply test/Conversion/intel/tritongpu_to_gen.mlir:1010]: done

File: test/Conversion/intel/tritongpu_to_gen.mlir
[the remark this answers is not in the document]
Comment (whitneywhtsang) [reply test/Conversion/intel/tritongpu_to_gen.mlir:1]: Yup, please file an issue for this problem.

[the remark this answers is not in the document]
Comment (whitneywhtsang) [reply test/Conversion/intel/tritongpu_to_gen.mlir:1]: CHECK lines for test_get_num_program appear to still be missing, any reasons for that?

[the remark this answers is not in the document]
Comment (anmyachev) [reply test/Conversion/intel/tritongpu_to_gen.mlir:1]: https://github.com/intel/intel-xpu-backend-for-triton/issues/1702

[the remark this answers is not in the document]
Comment (whitneywhtsang) [reply test/Conversion/intel/tritongpu_to_gen.mlir:1]: Actually let's remove `test_get_num_program` with `"triton_gpu.num-ctas" = 4 : i32`, as num-ctas > 1 is not really supported for intel backend now.

[the remark this answers is not in the document]
Comment (anmyachev) [reply test/Conversion/intel/tritongpu_to_gen.mlir:1]: done

>> CHECK lines for test_get_num_program appear to still be missing, any reasons for that?

Several tests have the same name, I just accidentally missed one :)

**Review by whitneywhtsang — APPROVED** [review]

**Comment by anmyachev** [comment]
@whitneywhtsang thanks for the review!
